randomness requirement (#273)
w3c/distributed-tracing: spec/50-privacy.md

## Privacy of traceparent field

> The `traceparent` field has a predefined set of values. These values are randomly-generated numbers. If a random number generator has any logic of using user identifiable information like IP address - this information may be exposed. Random number generators MUST NOT rely on any information that can potentially be user identifiable.

Reviewer: - `traceparent` doesn't have a predefinded set of values. This reads as if `traceparent` takes an enum.
- I've also tweaked the language around PII

```suggestion
The `traceparent` field is comprised of randomly-generated numbers. If a random number generator leverages any user identifiable information like IP address as seed state - this information may be exposed. Random number generators MUST NOT rely on any information that can potentially be user identifiable.
```

> Another privacy risk of the `traceparent` field is an ability to correlate calls made as part of a single transaction. A downstream service may track and correlate two or more calls made in a single transaction and make assumptions about the identity of the caller of one call base on information in another call.

Reviewer: Typo fix

```suggestion
Another privacy risk of the `traceparent` field is an ability to correlate calls made as part of a single transaction. A downstream service may track and correlate two or more calls made in a single transaction and make assumptions about the identity of the caller of one call based on information from another call.
```

> Note, both privacy concerns of `traceparent` field are theoretical rather than

Reviewer: ```suggestion
Note, both of the mentioned privacy concerns of the `traceparent` field are theoretical rather than
```

practical. Some services initiating or receiving a call MAY choose to restart a
`traceparent` field to eliminate those risks completely. It is recommended to

> find a way to minimize the number of <a>distributed trace</a> restarts to promote

Reviewer: ```suggestion
find a way to minimize the number of distributed trace restarts to promote
```

Author: this is needed to create a link to glossary

> interoperability of tracing vendors. Different techniques may be used. Services

Reviewer: ```suggestion
interoperability of tracing vendors. Instead, different techniques may be used. For example, services
```

> may define trust boundaries of upstream and downstream connections and level of

Reviewer: ```suggestion
may define trust boundaries of upstream and downstream connections and the level of
```

exposure any calls may bring. For instance, only restart `traceparent` for
authentication calls from or to external services.

Services may also define an algorithm and audit mechanism to validate randomness
of incoming or outgoing random numbers in the `traceparent` field. Note, this
algorithm will be services-specific and not a part of this specification. One

> example can be a temporal algorithm where a reversible hash function is applied

Reviewer: ```suggestion
example could be a temporal algorithm where a reversible hash function is applied
```

The `traceparent` field has a predefined set of values. These values are randomly-generated numbers. If a random number generator has any logic of using user-identifiable information like IP address - this information may be exposed. Random number generators MUST NOT rely on any information that can potentially be user-identifiable.

Another privacy risk of the `traceparent` field is an ability to correlate calls made as part of a single transaction. A downstream service may track and correlate two or more calls made in a single transaction and make assumptions about the identity of the caller of one call base on information in another call.

Note, both privacy concerns of `traceparent` field are theoretical rather than

> practical. Some service initiating or receiving a call MAY choose to restart a

Reviewer 2: ```suggestion
practical. Some services initiating or receiving a call MAY choose to restart a
```

`traceparent` field to eliminate those risks completely. It is recommended to

> find a way to minimize number of <a>distributed trace</a> restarts to promote

Reviewer 2: ```suggestion
find a way to minimize the number of <a>distributed trace</a> restarts to promote
```

interoperability of tracing vendors. Different techniques may be used. Services

> may define trust boundaries of upstream and downstream connections and level of

Reviewer 3: The "and level of exposure" part here doesn't sound correct. But maybe it's just my English that's not good enough

Author: I'm open for suggestions

Reviewer 2: I think it is correct. Alternatively, we can make a recommendation that we recommend checking egress traffic or calls to untrusted services.

exposure any calls may bring. For instance, only restart `traceparent` for
authentication calls from or to external services.

Services may also define an algorithm and audit mechanism to validate randomness

> of incoming or outgoing random numbers in traceparent `field`. Note, this

Reviewer 2: ```suggestion
of incoming or outgoing random numbers in the `traceparent` field. Note, this
```

algorithm will be services-specific and not a part of this specification. One

> example can be a temporal algorithms where reversible hash function is applied

Reviewer 3: ```suggestion
example can be a temporal algorithms where a reversible hash function is applied
```

> to the current clock time. So receiver can validate that the time is in agreed

Reviewer 3: ```suggestion
to the current clock time. So the receiver can validate that the time is within agreed upon
```
